adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,78 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Convert this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage: CoverageRDD = {
+    val coverage: RDD[(ReferenceRegion, Int)] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
+        .sortByKey()
+        .mapPartitions(i => merge(i)) // merge adjacent regions with the same coverage
```

Comment:
IMO, this (cache --> sort --> merge) should get factored out to `CoverageRDD`, and the `CoverageRDD` should have some sort of a `collapse` method? There are places where it doesn't necessarily make things simpler to have coverage collapsed down, e.g., https://github.com/bigdatagenomics/quinine/blob/master/quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/targeted/PanelStats.scala#L64-L101.

Also,